recipes/scri/meta.yaml

```
@@ -0,0 +1,50 @@
+{% set name = "scri" %}
+{% set version = "2019.3.21.21.27.29" %}
+{% set sha256 = "4abb5b8619ff204808c50c2402e993b695ab49f05841f38d75cea861d05aee09" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: package_version={{ version }} {{ PYTHON }} -m pip install --no-deps --ignore-installed . -vv           # [not win]
+  # script: set "package_version={{ version }}" && {{ PYTHON }} -m pip install --no-deps --ignore-installed . -vv  # [win]
+  skip: True  # [win]
```

Could this be a `noarch` package?
```suggestion
  noarch: python
```

---

Is it ok that windows doesn't pass in this case?

---

`noarch` packages are only build on Linux by default so it won't affect the feedstock

---

If this packaged has any kind of dependency in any platform this project cannot be ``noarch`` even if it is a ``python`` project

---

The only issue here is that there is a missing dependency on Windows (`spinsfast`). That will cause the CI to fail on staged-recipes but doesn't prevent it from being `noarch`.

---

Right, so I guess @moble  wanted the CIs here to pass, which is why he made it arch dependent.

---

That was my thinking, alright.  Even [the linter failed](https://github.com/conda-forge/staged-recipes/pull/8028#issuecomment-475462380) when I had it in there.  Are you saying that once it's a feedstock, it would build without complaint?  If so, would you be willing to merge this even with failing CIs?

---

Yes it will do and I'm happy to merge pull requests with failing CI in situations like this.

For the linter, you could remove the need for selectors by creating a `build.sh` (Linux/macOS) and `bld.bat` (Windows) and removing the `script` lines.

---

Sorry, now I'm confused.  This package has a hard dependency (can't even `import scri`) on `spinsfast`, which is not built for Windows.  So this package shouldn't be built for Windows.  Right?  Then what would go in `bld.bat`, just some `exit 1` type statement?

---

Basically, this package would consider itself as a no arch, fail to build on Windows, and fail to install on Windows due to a missing dependency. If that dependency ever gets built for windows, it will become automatically installable.

The recipe would simply look like windows would be possible to build. As such no windows specific would be used.

---

Alright, but I really need some help understanding exactly what you guys are suggesting.  Here's what I think you're suggesting.  Please correct me if I'm wrong.

  1. Remove `skip: True  # [win]`
  2. Add `noarch: python`
  3. Remove the `script:` lines
  4. Add a `bld.bat` file containing just `exit 1`
  5. Add a `build.sh` file containing `$PYTHON -m pip install --no-deps --ignore-installed . -vv`

Also, how do I get the `meta.yaml` variable `{{ version }}` as the environment variable `package_version` inside the `build.sh` script?  [This issue](https://github.com/conda/conda-build/issues/1553) and references therein seem to suggest this is not supported.

---

Keep the script lines, just remove the comment at the end of the first one

---

Sorry for the plug: If you use things like Miniver or versioneer, the zip from the **releases** on GitHub will not need you to pass the version parameter.